### DISCUSSION TITLE
requiring a list to be not empty

### DISCUSSION THREAD

**Discussion by Zariel** [cue-lang/cue#1761] [Q&A] [answered]
Been trying to figure out how to make a list be required to be set and not empty. Setting `myList: [...string]` seems to imply that the list can have 0 or more elements. It is possible to require a specific number of elements but I am yet to figure out if it is possible to require 1 or more.

**Comment by slewiskelly** [comment, marked as answer]
You can do either of the following:

```cue
myList: [string, ...string]
```

```cue
myList: [...string] & list.MinItems(1)
```
